packages/portal-proto/src/features/facets/CreateFacetCard.tsx

```diff
@@ -141,43 +165,70 @@ export const createFacetCard = (
         description={facet.description}
       />
     );
+  } else if (facet.facet_type === "set") {
+    return (
+      <SetFacet
+        key={`${idPrefix}-set-${facet.full}`}
+        facetName={facet?.name}
+        facetTitle={facet.title}
+        facetBtnToolTip={facet?.toolTip}
+        field={facet.full}
+        valueLabel={valueLabel}
+        hooks={dataFunctions as SetFacetHooks}
+      />
+    );
   }
   return <div> Unknown FacetType {facet.facet_type}</div>;
 };
 
+type CreateFacetCardFromListProps = Pick<
+  CreateFacetCardProps,
+  | "dataFunctions"
+  | "idPrefix"
+  | "valueLabel"
+  | "dismissCallback"
+  | "hideIfEmpty"
+  | "showPercent"
+  | "width"
+> & { facets: Partial<FacetCardDefinition>[]; facetNameSections?: number };
+
 /**
  * Creates and returns an array of Facet components defined by the facet definition array
- * @param facets - array of FacetDefinitions to create
- * @param dataFunctions - get/set hooks
- * @param valueLabel - string used to label counts
+ * @param facet - array of FacetDefinitions to create
+ * @param dataFunctions - data getter and setter hooks
+ * @param valueLabel - label for counts
  * @param idPrefix - prefix for created Facet Component key prop. This is used to ensure the ref
  *                  has a 1) unique 2) persistent id, so each call to createFacetCardsFromList must
  *                  have a unique prefix, the name of the analysis tool is a good choice
  * @param dismissCallback - define if facet should be removable from their parent
  * @param hideIfEmpty - hide facets if they do not have data
+ * @param showPercent - whether to show the count percent of whole
  * @param facetName - optional name of facet (if undefined it will be extracted from the full field name)
- * @param width - override the default width.
+ * @param width - override the default width
  */
-export const createFacetCardsFromList = (
-  facets: ReadonlyArray<FacetDefinition>,
-  dataFunctions: FacetRequiredHooks,
-  idPrefix: string,
-  valueLabel: string,
-  dismissCallback: (string) => void = undefined,
+
+export const createFacetCardsFromList = ({
+  facets,
+  dataFunctions,
+  idPrefix,
+  valueLabel,
+  dismissCallback = undefined,
   hideIfEmpty = false,
+  showPercent = true,
   facetNameSections = 1,
-  width?: string,
-): ReadonlyArray<React.ReactNode> => {
+  width = undefined,
+}: CreateFacetCardFromListProps): ReadonlyArray<React.ReactNode> => {
   return facets.map((x) =>
```

Comment:
can we name it `facet` as
```suggestion
  return facets.map((facet) =>
```

Reply:
Sure, good suggestion.